### DISCUSSION TITLE
Why is "partial" not handled inside the Modifiers production?

### DISCUSSION THREAD

**Discussion by vsajip** [javacc21/javacc21#217] [Q&A]
https://github.com/javacc21/javacc21/blob/af739f35c3a15fba3d21f4de579b4ef408351987/examples/csharp/CSharp.javacc#L56-L57
This sequence occurs quite a lot of times in this grammar.

**Comment by revusky** [comment]
It seems that "partial" is not exactly a modifier, like "public" and "static" and so forth. The various modifiers can appear in any order, so for example, you can write: "public static" or "static public", but "partial" has to appear after the various modifiers but before the type, like "class" or "interface" or whatever. I guess the point is to be able to put part of a class definition, say, into another source file, probably closer to where that part (methods etc) are actually used.

**Comment by revusky** [comment]
Of course, on second thought, I guess "partial" could be handled by the Modifiers production, the key point being that, assuming it's present, it would have to be the *last* modifier.
